File: app/src/main/java/chat/rocket/android/server/infraestructure/ServerDao.java
package chat.rocket.android.server.infraestructure;

import android.arch.persistence.room.Dao;
import android.arch.persistence.room.Delete;
import android.arch.persistence.room.Insert;
import android.arch.persistence.room.OnConflictStrategy;
import android.arch.persistence.room.Query;
import android.arch.persistence.room.Update;

import java.util.List;

import io.reactivex.Single;

@Dao
public interface ServerDao {
    @Insert(onConflict = OnConflictStrategy.FAIL)
    void insertServer(ServerEntity serverEntity);

    @Update
    void updateServer(ServerEntity serverEntity);

    @Delete
    void deleteServer(ServerEntity serverEntity);

    @Query("SELECT * FROM server")
    Single<List<ServerEntity>> getServers();

    @Query("SELECT * FROM server WHERE id = :serverId")
    Single<ServerEntity> getServer(Long serverId);
}

File: base/src/main/java/chat/rocket/android/server/infraestructure/ServerDao.kt
package chat.rocket.android.server.infraestructure

import android.arch.persistence.room.*
import io.reactivex.Single

@Dao
interface ServerDao {

    @Query("SELECT * FROM server")
    fun getServers(): Single<List<ServerEntity>>

    @Insert(onConflict = OnConflictStrategy.FAIL)
    fun insertServer(serverEntity: ServerEntity)

    @Update
    fun updateServer(serverEntity: ServerEntity)

    @Delete
    fun deleteServer(serverEntity: ServerEntity)

    @Query("SELECT * FROM server WHERE id = :serverId")
    fun getServer(serverId: Long?): Single<ServerEntity>
}
